### ISSUE TITLE
Use bash options to make scripts more robust 

### ISSUE THREAD

**Issue by chrisburr** [lhcb/analysis-essentials#42]
The bash lesson should recommend using some options at the top of scripts to make them easier to debug. My personal preference is to start scripts with

```bash
#!/bin/bash
set -euo pipefail
IFS=$'\n\t'
```

for the reasons listed [here](http://redsymbol.net/articles/unofficial-bash-strict-mode/).

**Comment by alexpearce** [comment]
Agreed that these are useful. I often prefer to use the long-hand version for each flag, as it's a little more self-documenting: (and Googleable)

```bash
set -o errexit -o nounset -o pipefail
```
On the other hand, these lines are quite a lot like magic, especially the `IFS` line. Are we willing to explain what they do in the lessons?
